ambry-store/src/test/java/com/github/ambry/store/StoreFindTokenTest.java

```diff
@@ -273,35 +347,109 @@ static DataInputStream getSerializedStream(StoreFindToken token, short version)
         bufWrap.putLong(logOffset);
         // add index start offset
         bufWrap.putLong(indexStartOffset);
-        // add storekey
-        if (key != null) {
-          bufWrap.put(key.toBytes());
+        // add storeKey
+        if (storeKeyInBytes.length > 0) {
+          bufWrap.put(storeKeyInBytes);
         }
         break;
       case StoreFindToken.VERSION_1:
-        sessionId = token.getSessionId();
-        key = token.getStoreKey();
-        byte[] offsetBytes = token.getOffset() != null ? token.getOffset().toBytes() : new byte[0];
+        byte[] offsetBytes = token.getOffsetInBytes();
         // version size + sessionId length size + session id size + type + log offset / index segment start offset size
         // + store key size
-        size = 2 + 4 + (sessionId == null ? 0 : sessionId.toString().getBytes().length) + 2 + offsetBytes.length + (
-            key == null ? 0 : key.sizeInBytes());
+        size = 2 + 4 + sessionIdBytes.length + 2 + offsetBytes.length + storeKeyInBytes.length;
         bytes = new byte[size];
         bufWrap = ByteBuffer.wrap(bytes);
         // add version
         bufWrap.putShort(StoreFindToken.VERSION_1);
         // add sessionId
-        Utils.serializeNullableString(bufWrap, sessionId == null ? null : sessionId.toString());
+        bufWrap.putInt(sessionIdBytes.length);
+        bufWrap.put(sessionIdBytes);
         // add type
-        type = token.getType();
         bufWrap.putShort((byte) type.ordinal());
         bufWrap.put(offsetBytes);
-        if (key != null) {
-          bufWrap.put(key.toBytes());
+        if (storeKeyInBytes.length > 0) {
```

Comment:
how is this method different from `StoreFindToken.toBytes`

Reply:
This was written to test deserialization of new token before production code supports serializing new token.